portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,96 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.sps.data.Message;
+import com.google.gson.Gson;    // To use gson function
 
-import java.io.IOException;
+// to use query capabilities
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+// to use datastore capabilities
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException; // to use exceptions
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello A!</h1>");
+    Query query = new Query("Mess").addSort("timestamp", SortDirection.DESCENDING);
+
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Message> messages = new ArrayList<>();
```

Comment:
No need to act on this, just wanted to show you a cool Java feature. I learned this at work and my mind was blown.

Java 8 introduced this very cool functional programming feature called lambdas, which enabled a second very cool programming paradigm called Streams. Using a stream, this expression could be rewritten as:

ArrayList comments = results.asIterable().stream().map(Message::fromEntity).collect(Collectors.toCollection(ArrayList::new));

(this is assuming you have the fromEntity static method in your Message class)

More about streams:
https://www.tutorialspoint.com/java8/java8_streams.htm

More about the colons (method reference):
https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html

More about lambdas:
https://www.geeksforgeeks.org/lambda-expressions-java-8/

Reply:
I will look more into this!